python/libkvikio/cmake/thirdparty/get_nvcomp.cmake

```
@@ -0,0 +1,35 @@
+# =============================================================================
+# Copyright (c) 2021-2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+set(KVIKIO_USE_PROPRIETARY_BINARY ON)
+
+# This function finds nvcomp and sets any additional necessary environment variables.
+function(find_and_configure_nvcomp)
+
+  include(${rapids-cmake-dir}/cpm/nvcomp.cmake)
+  set(export_args)
+  if(KvikIO_EXPORT_NVCOMP)
+    # We're vendoring nvcomp and we only want `libnvcomp.so.4`
+    set(export_args BUILD_EXPORT_SET nvcomp)
```

Should this be `INSTALL_EXPORT_SET`? (I don't know for sure.)
```suggestion
    set(export_args INSTALL_EXPORT_SET nvcomp)
```

---

I don't think so -- I can't say I really understand the difference, but `INSTALL_EXPORT_SET` balloons the size of the resulting wheel and includes many more copies of the nvcomp SO:

```
checking '/tmp/wheelhouse/libkvikio_cu12-25.6.0a23-py3-none-manylinux_2_28_x86_64.whl'
----- package inspection summary -----
file size
  * compressed size: 0.2G
  * uncompressed size: 0.5G
  * compression space saving: 67.8%
contents
  * directories: 22
  * files: 142 (7 compiled)
size by extension
  * .a - 0.1G (26.2%)
  * .so - 0.1G (25.1%)
  * .4 - 0.1G (24.3%)
  * .11 - 0.1G (24.3%)
  * .h - 0.6M (0.1%)
  * .hpp - 0.4M (0.1%)
  * no-extension - 63.4K (0.0%)
  * .cmake - 63.3K (0.0%)
  * .cuh - 12.4K (0.0%)
  * .py - 4.7K (0.0%)
  * .txt - 38.0B (0.0%)
largest files
  * (0.1G) libkvikio/lib64/libnvcomp_static.a
  * (0.1G) libkvikio/lib64/libnvcomp.so
  * (0.1G) libkvikio/lib64/libnvcomp.so.4
  * (0.1G) libkvikio/lib64/libnvcomp.so.4.2.0.11
  * (4.2M) libkvikio/lib64/libkvikio.so
------------ check results -----------
1. [distro-too-large-compressed] Compressed size 0.2G is larger than the allowed size (90.0M).
```

---

Effectively by setting `INSTALL_EXPORT_SET` you are causing all of nvcomp's install rules to be activated, whereas by not setting and using the wheelhelpers function you are just copying in the binaries.